### PR TITLE
fix(theme): making all component tokens optional

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

In the large theme type refactor @calebpollman and I did, there were some misses. Some component theme types were not marking all properties as optional, which causes some TS issues. This fixes those issues and adds a test to ensure this doesn't happen again

#### Description of how you validated changes

Tests pass (after making the test which failed)

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/__tests__/themeType.test.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Theme } from '../index';
+
+describe('Amplify UI Theme', () => {
+  describe('components', () => {
+    it('should allow empty components', () => {
+      const theme: Theme = {
+        name: 'test',
+        tokens: {
+          components: {
+            alert: {},
```
Comment (dbanksdesign) [inline packages/ui/src/theme/__tests__/themeType.test.ts:10]: Is there a better way to test a type shape like this? Also, if we add a component we will need to add it here. Grabbing all the component names and creating an object dynamically will not trigger a TS error.

File: packages/ui/src/theme/tokens/components/button.ts [hunk 2]
```diff
@@ -63,7 +63,8 @@ export type ButtonTokens<Output extends OutputVariantKey> =
     | 'borderWidth'
     | 'borderStyle'
     | 'borderRadius'
-    | 'color'
+    | 'color',
+    Output
```
Comment (dbanksdesign) [inline packages/ui/src/theme/tokens/components/button.ts:67]: The 2 types of issue are this one, where we forgot to pass the `Output` generic down, and the one below where we just forgot to add a `?`

File: packages/ui/src/theme/__tests__/themeType.test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Theme } from '../index';
+
+describe('Amplify UI Theme', () => {
+  describe('components', () => {
+    it('should allow empty components', () => {
+      const theme: Theme = {
```
[the remark this answers is not in the document]
Comment (ErikCH) [reply packages/ui/src/theme/__tests__/themeType.test.ts:6]: Not sure how we can skip this error...

**Review by ErikCH — DISMISSED** [review]
Looks good, not sure how we can exclude the unused variable warning.

**Review by joebuono — APPROVED** [review]

**Review by ErikCH — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by joebuono — APPROVED** [review]

**Pull request merged**